tests/test_dimensions.py

```diff
@@ -6,10 +6,10 @@
 def test_dimensions():
     dims = ansunits.BaseDimensions
     d1 = ansunits.Dimensions(dimensions_container={})
-    assert d1.dimensions == {}
+    assert d1._dimensions == {}
 
     d2 = ansunits.Dimensions(dimensions_container={dims.MASS: 1.0, dims.LENGTH: -2.0})
-    assert d2.dimensions == {dims.MASS: 1.0, dims.LENGTH: -2.0}
+    assert d2._dimensions == {dims.MASS: 1.0, dims.LENGTH: -2.0}
```

Comment:
Since you made this member protected probably we should not unit test it?  i.e. it's not technically part of the public api anymore.

Reply:
Should I re-add the dimensions property to expose the dict, to test that it initializes properly? Would there be a cleaner way to test this?

Reply:
We are able to iterate over the `Dimensions` object, so we can build a `dict` in that process (dictionary comprehension) and compare the result with the expected `dict`.  We don't need to expose `_dimensions`.

Reply:
Just in case, it's: `{k:v for k,v in d2}`.

Reply:
Just to mention that `Dimensions` is iterable but not subscriptable, nor is the internal `dict` truly accessible. I think that's all fine from my point of view.